src/core/api/media_element_track_choice_manager.ts

```diff
@@ -0,0 +1,468 @@
+
+  private _audioTracks : Array<{ track: ITMAudioTrack; origTrack: AudioTrack }>;
+  private _textTracks : Array<{ track: ITMTextTrack; origTrack: TextTrack }>;
+  private _videoTracks : Array<{ track: ITMVideoTrack; origTrack: VideoTrack }>;
+
+  private _mediaElement : HTMLMediaElement;
+
+  constructor(
+    defaults : { preferredAudioTracks : BehaviorSubject<IAudioTrackPreference[]>;
+                 preferredTextTracks : BehaviorSubject<ITextTrackPreference[]>; },
+    mediaElement: HTMLMediaElement
+  ) {
+    super();
+    const { preferredAudioTracks, preferredTextTracks } = defaults;
+
+    this._preferredAudioTracks = preferredAudioTracks;
+    this._preferredTextTracks = preferredTextTracks;
+
+    this._mediaElement = mediaElement;
+
+    /**
+     * Check if track array is different from an other one
+     * @param {Array.<Object>} oldTrackArray
+     * @param {Array.<Object>} newTrackArray
+     * @returns {boolean}
+     */
+    function areTrackArraysDifferent(
+      oldTrackArray: Array<{ origTrack: VideoTrack|AudioTrack|TextTrack }>,
+      newTrackArray: Array<{ origTrack: VideoTrack|AudioTrack|TextTrack }>
+    ) {
+      if (newTrackArray.length !== oldTrackArray.length) {
+        return true;
+      }
+      for (let i = 0; i < newTrackArray.length; i++) {
+        if (newTrackArray[i].origTrack !== oldTrackArray[i]?.origTrack) {
+          return true;
+        }
+      }
+      return false;
+    }
+
+    // Create audio tracks from native audio tracks.
+    const createAudioTracks = (): Array<{ track: { id: string;
+                                                   normalized: string;
+                                                   language: string;
+                                                   audioDescription: boolean; };
+                                          origTrack: AudioTrack; }> => {
+      const newAudioTracks = [];
+      const { audioTracks } = mediaElement;
+      const languagesOccurences: Partial<Record<string, number>> = {};
+      for (let i = 0; i < audioTracks.length; i++) {
+        const audioTrack = audioTracks[i];
+        const language = audioTrack.language === "" ? "nolang" :
+                                                      audioTrack.language;
+        const occurences = languagesOccurences[language] ?? 1;
+        const id = "gen_audio_" +
+                   language +
+                   "_" +
+                   occurences.toString();
+        languagesOccurences[language] = occurences + 1;
+        const track = { language: audioTrack.language,
+                        id,
+                        normalized: normalizeLanguage(audioTrack.language),
+                        audioDescription: false };
+        newAudioTracks.push({ track,
+                              origTrack: audioTrack });
+      }
+      return newAudioTracks;
+    };
+
+    this._audioTracks = [];
+    if (mediaElement.audioTracks != null) {
```

Comment:
`null`? `undefined`? or both?

Maybe we should be more explicit in any case

Reply:
Done